tests/Fixtures/DocBlockType/PhpstanMultipleArrayShapes.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JMS\Serializer\Tests\Fixtures\DocBlockType;
+
+/**
+ * @phpstan-type Settings array{
+ *  method: string,
+ *  amount: array{type: string, value: string|null}
+ * }
+ * @phpstan-type Details array{
```

Comment:
Can we add also test & support for some types like: `array<int, SomeType>` or `array<SomeType>`, please? It is already supported by Serializer, so I believe it should be also supported for resolving types here.

Reply:
Ok, I've added support array collections  :)